src/theory/bags/normal_form.cpp

```diff
@@ -564,29 +564,13 @@ Node NormalForm::evaluateChoose(TNode n)
   Assert(n.getKind() == BAG_CHOOSE);
   // Examples
   // --------
-  // - (choose (emptyBag String)) = "" // the empty string which is the first
-  //   element returned by the type enumerator
-  // - (choose (MK_BAG "x" 4)) = "x"
-  // - (choose (union_disjoint (MK_BAG "x" 4) (MK_BAG "y" 1))) = "x"
-  //     deterministically return the first element
-
-  if (n[0].getKind() == EMPTYBAG)
-  {
-    TypeNode elementType = n[0].getType().getBagElementType();
-    TypeEnumerator typeEnumerator(elementType);
-    // get the first value from the typeEnumerator
-    Node element = *typeEnumerator;
-    return element;
-  }
+  // - (bag.choose (MK_BAG "x" 4)) = "x"
 
   if (n[0].getKind() == MK_BAG)
   {
     return n[0][0];
   }
-  Assert(n[0].getKind() == UNION_DISJOINT);
```

Comment:
I think you can additionally return the first element of UNION_DISJOINT if the argument is constant.  This is important since this would ensure we have an evaluator for CHOOSE.